src/ai/backend/web/server.py

```diff
@@ -527,20 +543,32 @@ async def token_login_handler(request: web.Request) -> web.Response:
             # in token-based login. Each authorize hook plugin will deal with various type of
             # `sToken` and related parameters to authorize a user. In this process, email and
             # password do not play any role.
-            token = await api_session.User.authorize(
+            auth_result = await api_session.User.authorize(
                 "fake-email", "fake-pwd", extra_args=extra_args
             )
+            match auth_result:
+                case AuthSuccessResponse():
+                    token = auth_result
+                case RequireTOTPRegistrationResponse():
+                    result["authenticated"] = False
+                    result["data"] = {
+                        "type": "https://api.backend.ai/probs/require-totp-registration",
+                        "title": "Require TOTP Registration",
+                        "details": "The user must register TOTP.",
+                        "totp_registration_token": auth_result.token,
+                    }
+                    return web.json_response(result)
             stored_token = {
                 "type": "keypair",
-                "access_key": token.content["access_key"],
-                "secret_key": token.content["secret_key"],
-                "role": token.content["role"],
-                "status": token.content.get("status"),
+                "access_key": token.access_key,
+                "secret_key": token.secret_key,
+                "role": token.role,
+                "status": token.status,
             }
             public_return = {
-                "access_key": token.content["access_key"],
-                "role": token.content["role"],
-                "status": token.content.get("status"),
+                "access_key": token.access_key,
+                "role": token.role,
+                "status": token.status,
             }
             session["authenticated"] = True
             session["token"] = stored_token  # store full token
```

Comment:
The logic from the `stored_token` is the same as when we received the `AuthSuccessResponse`, which works fine for now, but the block separation doesn't look as clean.

Reply:
hmm, then how about raising an error and catch it?
Like this block https://github.com/lablup/backend.ai/blob/cfaec702deda14cb1dd77f42040797abcac09f21/src/ai/backend/web/server.py#L425-L435